File: src/presets/full.ts
import {
  StatusLine, GitGroup, Group, GitBranch, GitInsertions, GitDeletions,
  ContextGroup, ContextTokens, ContextSize, ContextPercent, Literal,
  ModelName, CostUSD, SessionDuration, SessionLinesAdded, SessionLinesRemoved,
} from '../dsl/index.js';
import type { SegmentNode } from '../types.js';

export const fullPreset: SegmentNode[] = StatusLine(() => [
  { type: 'pwd.smart', provider: 'pwd', style: { color: '31' } },
  { type: 'pwd.name', provider: 'pwd', style: { color: '39', bold: true } },
  GitGroup({ prefix: ' | ', color: '240' })(() => [
    GitBranch({ color: 'whiteBright', bold: true, prefix: '\ue0a0 ' }),
    Group({ prefix: ' [', suffix: ']' })(() => [
      GitInsertions({ color: 'green', prefix: '+' }),
      GitDeletions({ color: 'red', prefix: ' -' }),
    ]),
  ]),
  { type: 'literal', props: { text: ' | ' }, style: { color: '240' } },
  ModelName({ color: '240' }),
  { type: 'literal', props: { text: ' · ' } },
  ContextGroup()(() => [
    ContextTokens({ color: 'white', bold: true }),
    Literal({ text: '/' }),
    ContextSize({ color: 'white' }),
    Literal({ text: ' (' }),
    ContextPercent({ color: 'white' }),
    Literal({ text: ')' }),
  ]),
  { type: 'literal', props: { text: ' · ' } },
  CostUSD({ color: 'yellow', bold: true }),
  { type: 'literal', props: { text: ' · ' } },
  SessionDuration({ color: 'magenta' }),
  { type: 'literal', props: { text: ' · ' } },
  Group()(() => [
    SessionLinesAdded({ color: 'green', prefix: '+' }),
    SessionLinesRemoved({ color: 'red', prefix: ' -' }),
  ]),
]);

File: internal/preset/full.go
package preset

import "github.com/jheddings/ccnow/internal/types"

func fullPreset() []types.SegmentNode {
	return []types.SegmentNode{
		{Type: "pwd.smart", Provider: "pwd", Style: &types.StyleAttrs{Color: "31"}},
		{Type: "pwd.name", Provider: "pwd", Style: &types.StyleAttrs{Color: "39", Bold: true}},
		{
			Type:  "group",
			Style: &types.StyleAttrs{Prefix: " | ", Color: "240"},
			Children: []types.SegmentNode{
				{Type: "git.branch", Provider: "git", Style: &types.StyleAttrs{Color: "whiteBright", Bold: true, Prefix: "\ue0a0 "}},
				{Type: "git.insertions", Provider: "git", Style: &types.StyleAttrs{Color: "green", Prefix: " \u00b7 +"}},
				{Type: "git.deletions", Provider: "git", Style: &types.StyleAttrs{Color: "red", Prefix: " -"}},
			},
		},
		{Type: "model.name", Provider: "model", Style: &types.StyleAttrs{Prefix: " | "}},
		{
			Type:  "group",
			Style: &types.StyleAttrs{Prefix: " \u00b7 "},
			Children: []types.SegmentNode{
				{Type: "context.tokens", Provider: "context", Style: &types.StyleAttrs{Color: "white", Bold: true}},
				{Type: "context.size", Provider: "context", Style: &types.StyleAttrs{Color: "white", Prefix: "/"}},
				{Type: "context.percent", Provider: "context", Style: &types.StyleAttrs{Color: "white", Prefix: " (", Suffix: ")"}},
			},
		},
		{Type: "cost.usd", Provider: "cost", Style: &types.StyleAttrs{Color: "yellow", Bold: true, Prefix: " \u00b7 "}},
		{Type: "session.duration", Provider: "session", Style: &types.StyleAttrs{Color: "magenta", Prefix: " \u00b7 "}},
		{
			Type:  "group",
			Style: &types.StyleAttrs{Prefix: " \u00b7 "},
			Children: []types.SegmentNode{
				{Type: "session.lines-added", Provider: "session", Style: &types.StyleAttrs{Color: "green", Prefix: "+"}},
				{Type: "session.lines-removed", Provider: "session", Style: &types.StyleAttrs{Color: "red", Prefix: " -"}},
			},
		},
	}
}
